internal/commands/user/delete.go

```diff
@@ -0,0 +1,152 @@
+	cmd.inputs.Flags(fs)
+	fs.StringSliceVarP(&cmd.inputs.Users, flagUser, flagUserShort, []string{}, flagUserDeleteUsage)
+	fs.VarP(
+		flags.NewEnumSet(&cmd.inputs.ProviderTypes, validProviderTypes),
+		flagProvider,
+		flagProviderShort,
+		flagProviderUsage,
+	)
+	fs.VarP(&cmd.inputs.State, flagState, flagStateShort, flagStateUsage)
+	fs.VarP(&cmd.inputs.Status, flagStatus, flagStatusShort, flagStatusUsage)
+}
+
+// Inputs is the command inputs
+func (cmd *CommandDelete) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Setup is the command setup
+func (cmd *CommandDelete) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = realm.NewAuthClient(profile)
+	return nil
+}
+
+// Handler is the command handler
+func (cmd *CommandDelete) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, appErr := app.Resolve(ui, cmd.realmClient, cmd.inputs.Filter())
+	if appErr != nil {
+		return appErr
+	}
+
+	users, err := cmd.inputs.ResolveUsers(ui, cmd.realmClient, app)
+	if err != nil {
+		return err
+	}
+
+	for _, user := range users {
+		deleteErr := cmd.realmClient.DeleteUser(app.GroupID, app.ID, user.ID)
+		cmd.outputs = append(cmd.outputs, userOutput{user: user, err: deleteErr})
+	}
+	return nil
+}
+
+// Feedback is the command feedback
+func (cmd *CommandDelete) Feedback(profile *cli.Profile, ui terminal.UI) error {
+	if len(cmd.outputs) == 0 {
+		return ui.Print(terminal.NewTextLog("No users to delete"))
+	}
+
+	var outputByProviderType = map[string][]userOutput{}
+	for _, output := range cmd.outputs {
+		for _, identity := range output.user.Identities {
+			outputByProviderType[identity.ProviderType] = append(outputByProviderType[identity.ProviderType], output)
+		}
+	}
+	logs := make([]terminal.Log, len(outputByProviderType))
+	logIndex := 0
+	for providerType, outputs := range outputByProviderType {
+
+		sort.Slice(outputs, getUserOutputComparerBySuccess(outputs))
+
+		logs[logIndex] = terminal.NewTableLog(
+			fmt.Sprintf("Provider type: %s", providerType),
+			userDeleteTableHeaders(providerType),
+			userDeleteTableRows(providerType, outputs)...,
+		)
+		logIndex++
+	}
+	return ui.Print(logs...)
+}
+
+func userDeleteTableHeaders(providerType string) []string {
+	var headers []string
+	switch providerType {
+	case providerTypeAPIKey:
+		headers = append(headers, headerName)
+	case providerTypeLocalUserPass:
+		headers = append(headers, headerEmail)
+	}
+	headers = append(
+		headers,
+		headerID,
+		headerType,
+		headerDeleted,
+		headerDetails,
+	)
+	return headers
+}
+
+func userDeleteTableRows(providerType string, outputs []userOutput) []map[string]interface{} {
+	userDeleteTableRows := make([]map[string]interface{}, 0, len(outputs))
+	for _, output := range outputs {
+		userDeleteTableRows = append(userDeleteTableRows, userDeleteTableRow(providerType, output))
+	}
+	return userDeleteTableRows
+}
+
+func userDeleteTableRow(providerType string, output userOutput) map[string]interface{} {
+	var msg string
```

Comment:
can we align this name with the header const name associated with it?  can use `var details string` here, or something that achieves the same?  (if you go the "message" route, i'd avoid using `msg` at that point)